rebasehelper/tests/test_utils.py

```diff
@@ -42,6 +43,12 @@
 from rebasehelper.utils import LookasideCacheHelper
 
 
+class TestGitHelper(object):
+    def test_get_user_and_email(self):
+        assert GitHelper.get_user()
+        assert GitHelper.get_email()
```

Comment:
What's the point of this test? In what situation should it fail? I don't think I get it.

Reply:
Sorry it may be meaningless.
Actually I want to test above cases that I tested.
But I was not sure how much we implement the unit tests in rebase-helper.

Do you think we need additional unit tests for that in this case?

Reply:
If you want to test whether `git config --get user.name` works as expected, you should first configure it (e.g. like [here](https://github.com/rebase-helper/rebase-helper/blob/master/rebasehelper/application.py#L467)) and then check if `GitHelper.get_user()` returns what it should. You could configure it system-wide and then on user level and test both cases.
Where do you personally have it configured btw? As it didn't work with `--global` argument. You should test for such case too.

Reply:
@nforro thanks for the information. I will do it.

My configuration is

`~/.gitconfig` is this file.
> https://github.com/junaruga/my-config-pub/blob/master/home/.gitconfig
> [include]
> path = .gitconfig.local

And

```
$ cat ~/.gitconfig.local
[user]
    email = jaruga@redhat.com
    name = Jun Aruga
```

Reply:
Interesting, so if `include.path` is used, `--global` option doesn't work. I wonder whether that's a bug.

Reply:
Aha, it works with `--include` agument. So `git config --global --include --get user.name`. But I'm not sure whether it's better to drop `--global` or keep it and add `--include`. I can't think of any downsides of not using `--global` at the moment.

Reply:
I was also success.

```
$ git config --global --includes --get user.name
Jun Aruga

$ echo $?
0
```

I found below explanation.

```
$ man git-config
...
       --[no-]includes   
           Respect include.*  directives in config files when looking up values. Defaults
           to off when a specific file is given (e.g., using --file, --global, etc) and on
           when searching all config files.
```

I prefer `git config --get user.name` than `git config --global --includes --get user.name`.

Because `git config --get user.name` can decide to get the value by following git's default priority. And it covers all the below cases. In my experience, I used to manage Git name and email by local scope (`$(repo_path)/.git/config`).

* `--global` (`~/.gitconfig`)
* `--system` (`$(prefix)/etc/gitconfig`)
* `--local` (`$(repo_path)/.git/config`)

Reply:
I have got a long vacation until today. I would start working again from today. :)